File: frontend/src/types/Task.ts
export interface Task {
  id: number;
  title: string;
  description?: string;
  status: TaskStatus;
  createdAt: string;
  updatedAt: string;
}

export enum TaskStatus {
  TODO = 'TODO',
  IN_PROGRESS = 'IN_PROGRESS',
  IN_REVIEW = 'IN_REVIEW',
  DONE = 'DONE'
}

export interface TaskCreateDto {
  title: string;
  description?: string;
}

export interface TaskUpdateDto {
  title?: string;
  description?: string;
  status?: TaskStatus;
}

export interface User {
  id: number;
  name: string;
  email: string;
  lastLogin: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface UserUpdateDto {
  name?: string;
  email?: string;
}

export interface UserUpdateResponse {
  user: User;
  token: string;
}

File: backend/src/main/kotlin/com/example/kanban/model/Task.kt
package com.example.kanban.model

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "tasks")
data class Task(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,
    
    @Column(nullable = false)
    val title: String,
    
    @Column(columnDefinition = "TEXT")
    val description: String? = null,
    
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    val status: TaskStatus = TaskStatus.TODO,
    
    @Column(nullable = false, name = "user_id")
    val userId: Long,
    
    @Column(nullable = false)
    val createdAt: LocalDateTime = LocalDateTime.now(),
    
    @Column(nullable = false)
    val updatedAt: LocalDateTime = LocalDateTime.now()
)

enum class TaskStatus {
    TODO, IN_PROGRESS, IN_REVIEW, DONE
}